devito/interfaces.py

```diff
@@ -255,3 +265,60 @@ def indices(cls, shape):
         """
         _indices = [t, x, y, z]
         return _indices[:len(shape) + 1]
+
+
+class MemmapManager():
+    """Class for managing all memmap related settins"""
+    # used to enable memmap as default
+    _use_memmap = False
+    # flag for registering exit func
+    _registered = False
+    _default_disk_path = "/tmp/devito_disk"
+    # contains str name of all memmap file created
+    _created_data = set()
+    _default_exit_code = 0
+
+    @staticmethod
+    def memmap_enabled_by_default():
+        """call this method to enable memmap by default"""
+        return MemmapManager._use_memmap
+
+    @staticmethod
+    def set_default_disk_path(default_disk_path):
+        """call this method to change the default disk path for memmap"""
+        MemmapManager._default_disk_path = default_disk_path
+
+    @staticmethod
+    def setup(data_self, *args, **kwargs):
+        """call this method with arguments self, args, kwargs to setup memmap parameters"""
```

Comment:
You really need to document the arguments in `kwargs` this responds to in the docstring!